### PR TITLE
Remove ClientError

### PR DESCRIPTION
**Pull request by jeffallen**
Also: export some named errors that callers can check if they need to.

### CODE REVIEW

File: pop/app.go [hunk 1]
```diff
@@ -123,7 +123,7 @@ func orgLink(c *cli.Context) error {
 	addr := network.NewTCPAddress(fmt.Sprintf("%s:%s", addrs[0], port))
 	pin := c.Args().Get(1)
 	if err := client.PinRequest(addr, pin, cfg.OrgPublic); err != nil {
-		if err.ErrorCode() == service.ErrorWrongPIN && pin == "" {
+		if err == service.ErrorWrongPIN && pin == "" {
```
Comment (ineiti) [inline pop/app.go]: That works without using the `.Error()`?

File: randhound/randhound.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ import (
 	"github.com/dedis/onet"
 	"github.com/dedis/onet/log"
 	"github.com/dedis/onet/network"
+	"gopkg.in/dedis/crypto.v0/abstract"
```
Comment (ineiti) [inline randhound/randhound.go]: that's probably wrong here

File: randhound/proof.go [hunk 3]
```diff
@@ -9,6 +9,7 @@ import (
 	"github.com/dedis/kyber/poly"
 	"github.com/dedis/kyber/util/random"
 	"github.com/dedis/onet/crypto"
+	"gopkg.in/dedis/crypto.v0/abstract"
```
Comment (ineiti) [inline randhound/proof.go]: that's probably wrong here

File: randhound/randhound.go (hunk 2 above)
Comment (jeffallen) [reply randhound/randhound.go]: This is dead code that is waiting to be revived. I thought you were working on that?

File: pop/app.go (hunk 1 above)
Comment (jeffallen) [reply pop/app.go]: No, I mean I have taken care of that correction.

File: pop/service/api.go [hunk 4]
```diff
@@ -57,17 +37,17 @@ func NewClient() *Client {
...
 }
 
 // StoreConfig sends the configuration to the conode for later usage.
-func (c *Client) StoreConfig(dst network.Address, p *PopDesc, priv kyber.Scalar) onet.ClientError {
+func (c *Client) StoreConfig(dst network.Address, p *PopDesc, priv kyber.Scalar) error {
 	si := &network.ServerIdentity{Address: dst}
 	sg, e := schnorr.Sign(cothority.Suite, priv, p.Hash())
 	if e != nil {
-		return onet.NewClientError(e)
+		return e
```
Comment (kc1212) [inline pop/service/api.go]: nitpick: can we change it to `err` to be consistent?

**Pull request merged**